### PR TITLE
transformers.image_transforms.normalize wrong types

### LINKED ISSUE #35772: transformers.image_transforms.normalize documents and checks for the wrong type for std and mean arguments
### System Info

- `transformers` version: 4.49.0.dev0
- Platform: Linux-6.8.0-51-generic-x86_64-with-glibc2.39
- Python version: 3.11.11
- Huggingface_hub version: 0.27.1
- Safetensors version: 0.5.1
- Accelerate version: 1.2.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.5.1+cu124 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: Neither
- Using GPU in script?: No
- GPU type: NVIDIA GeForce GTX 1060 6GB

### Who can help?

_No response_

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
import numpy as np
from transformers.image_transforms import normalize

mock_img = np.random.rand(3, 10, 10)

# Simulating getting the std and mean from text
std = map(float, "1.5 0. 2.".split())
mean = map(float, "1.5 0. 2.".split())
# According to the doc, this setup should work, but it does not since "map" does not implement `len`

normalize(mock_img, std, mean)
```

### Expected behavior

Use the `Sequence` type instead of Iterable to let users know that `len` needs to be implemented.

### PR DESCRIPTION
**Pull request by CalOmnie**
# What does this PR do?

`transformers.image_transforms.normalize` types `std` and `mean` as Iterable, and checks for that, but calls `len` on them. The proper type for a sized iterable is `Sequence`, this PR adds that.

Fixes #35772

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by Rocketknight1 — APPROVED** [review]
Yes, this is correct. Thanks for the fix!

**Comment by tomasruizt** [comment]
This PR was probably supposed to be a refactor, and should not change the execution paths of the code. However, it turns out that numpy arrays are Iterables but not Sequences. And client code that used to pass numpy arrays for `mean` and `std` fails now. We found this because it broke our MiniCPM video-chat pipeline. Case here: https://huggingface.co/openbmb/MiniCPM-V-2_6/discussions/53

```python
from typing import Iterable, Sequence
import numpy as np

x = np.ones(3)
print(isinstance(x, Iterable))
#> True
print(isinstance(x, Sequence))
#> False
```

I can see that the type hint does not include numpy arrays, but passing a numpy array (which I guess is the intention of the type hint `Sequence[float]`  /  `Iterable[float]`) is reasonable, too. Would you say that client code is responsible for adjusting? Or could we revert? Or what would guys you suggest? 

Best :)

**Comment by CalOmnie** [comment]
Oh, you're right, I went one level too high in the typing, the type needed was `Collection` not `Sequence`, see [here](https://docs.python.org/3/library/collections.abc.html#collections.abc.Container)

image

I will have a PR out tonight unless someone beats me to it.

**Comment by CalOmnie** [comment]
Alright I created a fix and modified some tests to make sure it worked [here](https://github.com/huggingface/transformers/pull/36301)
